extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java

```diff
@@ -234,6 +234,7 @@ private void registerReflectionForApiResponseSchemaSerialization(BuildProducer<R
 
     @BuildStep(loadsApplicationClasses = true)
```

Comment:
This is already using the correct TCCL due to loadsApplicationClasses= true

Reply:
I wasn't aware of that flag. So which of the two solutions is preferred?
Asking because people are definitely going to be copying this kind of thing in other extensions

Reply:
loadsApplicationClasses is should be preferred, as it automatically sets the correct TCCL. If you are loading classes from the DeploymentClassLoader but have not set it to the TCCL there is the possibility for subtle bugs if the classes being loaded use the TCCL.

I did actually think about defaulting to true and requiring build steps that feed into generating the DeploymentClassLoader to set it to false but I am not sure how invasive that change would be.

Reply:
Gotcha, thanks. I'll update the PR accordingly